app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/PreferenceManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.kks.nimblesurveyjetpackcompose.util
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import com.google.gson.Gson
+import com.kks.nimblesurveyjetpackcompose.R
+
+const val PREF_ACCESS_TOKEN = "access_token"
+const val PREF_REFRESH_TOKEN = "refresh_token"
+const val PREF_LOGGED_IN = "logged_in"
+
+class PreferenceManager(context: Context) {
+    private val preferences: SharedPreferences =
+        context.getSharedPreferences(
+            "pref_${context.getString(R.string.app_name)}",
+            Context.MODE_PRIVATE
+        )
+
+    fun setStringData(key: String, value: String) {
+        preferences.edit { putString(key, value) }
+    }
+
+    fun setIntegerData(key: String, value: Int) {
+        preferences.edit { putInt(key, value) }
+    }
+
+    fun setLongData(key: String, value: Long) {
+        preferences.edit { putLong(key, value) }
+    }
+
+    fun setBooleanData(key: String, value: Boolean) {
+        preferences.edit { putBoolean(key, value) }
+    }
+
+    fun <T> setList(key: String?, list: List<T>?) {
+        val gson = Gson()
+        val json = gson.toJson(list)
+        preferences.edit().putString(key, json).apply()
+    }
```

Comment:
We could update this method to use Moshi instead, or get rid of it entirely since its not being used 🤔